src/vtok_agent/src/imds.rs

```diff
@@ -13,6 +13,8 @@ lazy_static! {
     static ref IMDS_CACHE: Arc<Mutex<Option<ImdsCache>>> = Arc::new(Mutex::new(None));
 }
 
+const INSTANCE_ROLE_ARN_TAG: &str = "InstanceRoleArn";
```

Comment:
Would it be possible to make it configurable rather then hardcoded?

Reply:
Checked together it won't be so trivial to pass configuration to a singleton of `ImdsCache` instance with current design. Let's keep it as is.